components/kfp/Run_component/component.py

```diff
@@ -0,0 +1,47 @@
+from typing import NamedTuple
+
+
+def run_component_or_pipeline(
+    component_url: 'Url',
+    arguments: dict,
+    endpoint: str = None,
+    wait_timeout_seconds: float = None,
+) -> NamedTuple('Outputs', [
+    ('run_id', str),
+    ('run_object', 'JsonObject'),  # kfp.ApiRunDetails
+]):
+    import json
+    import os
+    import kfp
+    from kfp_server_api import ApiClient
+    print('Loading component...')
+    op = kfp.components.load_component_from_url(component_url)
+    print('Loading component done.')
+    print('Submitting run...')
+    if not endpoint:
+        endpoint = os.environ['ML_PIPELINE_SERVICE_HOST'] + ':' + os.environ['ML_PIPELINE_SERVICE_PORT']
+    create_run_result = kfp.Client(host=endpoint).create_run_from_pipeline_func(op, arguments=arguments)
+    run_id = str(create_run_result.run_id)
+    print('Submitted run: ' + run_id)
+    print('Waiting for the run to finish...')
```

Comment:
I think it'd be very useful emitting another output, which is UI url of the pipeline run details page.

Reply:
How would you build that URL?

Reply:
https://github.com/kubeflow/pipelines/blob/fc769b82a752b5680a524bd4e488e634ef04aa23/v2/test/samples/fail.py#L48

I pass the real hostname into the component

Reply:
Done